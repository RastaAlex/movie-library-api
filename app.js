import express from 'express';
import bodyParser from 'body-parser';
import cors from 'cors';
import helmet from 'helmet';
import rateLimit from 'express-rate-limit';
import dotenv from 'dotenv';
import movieRoutes from './src/routes/moviesRoutes.js';
import userRoutes from './src/routes/usersRoutes.js';
import { sequelize } from './src/config/database.js';

dotenv.config();
const app = express();

const limiter = rateLimit({
  windowMs: 15 * 60 * 1000,
  max: 100,
});

app.use(limiter);

app.use(cors());
app.use(helmet());
app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: true }));

app.use(userRoutes);
app.use(movieRoutes);

const PORT = process.env.PORT || 8000;

const startServer = async () => {
  try {
    await sequelize.sync();
    app.listen(PORT, () => {
      console.log(`Server is running on port ${PORT}`);
    });
  } catch (error) {
    console.error('Unable to connect to the database:', error);
  }
};

if (process.env.NODE_ENV !== 'test') {
  startServer();
}

export { app };
