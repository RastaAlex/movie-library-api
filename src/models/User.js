import Sequelize from 'sequelize';
import { sequelize } from '../config/database.js';

const { DataTypes } = Sequelize;

const User = sequelize.define(
  'User',
  {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },
    email: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true,
    },
    name: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    password: {
      type: DataTypes.STRING,
      allowNull: false,
    },
  },
  {
    timestamps: false,
  },
);

export default User;
