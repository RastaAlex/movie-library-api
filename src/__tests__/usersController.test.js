import supertest from 'supertest';
import { app } from '../../app.js';
import { sequelize } from '../config/database.js';
import User from '../models/user.js';

let server;
let request;

beforeAll(async () => {
  await sequelize.sync({ force: true });
  server = app.listen(0);
  request = supertest(server);
});

afterAll(async () => {
  await sequelize.close();
  await server.close();
});

describe('Users API', () => {
  beforeEach(async () => {
    await User.destroy({ where: {}, truncate: true });
  });

  test('POST should create a new user', async () => {
    const newUser = {
      email: 'test@example.com',
      name: 'Test User',
      password: 'password123',
      confirmPassword: 'password123',
    };

    const response = await request.post('/api/v1/users').send(newUser);

    expect(response.statusCode).toBe(201);
    expect(response.body).toMatchSnapshot();
  });

  test('POST should return a token for an existing user', async () => {
    const newUser = {
      email: 'test@example.com',
      name: 'Test User',
      password: 'password123',
      confirmPassword: 'password123',
    };
    await request.post('/api/v1/users').send(newUser);

    const response = await request.post('/api/v1/sessions').send({
      email: 'test@example.com',
      password: 'password123',
    });

    expect(response.statusCode).toBe(200);
    expect(response.body).toMatchSnapshot();
  });
});
