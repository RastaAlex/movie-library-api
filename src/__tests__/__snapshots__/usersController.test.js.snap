// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users API POST should create a new user 1`] = `
{
  "status": 1,
  "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiZW1haWwiOiJ0ZXN0QGV4YW1wbGUuY29tIiwibmFtZSI6IlRlc3QgVXNlciIsImlhdCI6MTY4MTQ3MzM4OCwiZXhwIjoxNjgxNDc2OTg4fQ.cbizIjxhThYWH6j3kZueJ2Ds4NgHdCXKgfNWrTpqeck",
}
`;

exports[`Users API POST should return a token for an existing user 1`] = `
{
  "status": 1,
  "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MiwiZW1haWwiOiJ0ZXN0QGV4YW1wbGUuY29tIiwiaWF0IjoxNjgxNDczMzg5LCJleHAiOjE2ODE0NzY5ODl9.PRtDmsT0ZVOsmlDMAWrfwyJAPzpzlKXOo9xXdNe4_Lw",
}
`;
